### PR TITLE
add a couple missing stubs and fix an incorrect return type for redis stubs

### PR DESCRIPTION
**Pull request by brianmaissy**
relates to #3544

### CODE REVIEW

File: third_party/2and3/redis/client.pyi [hunk 1]
```diff
@@ -80,6 +80,8 @@ class Redis(object):
     def client_list(self): ...
     def client_getname(self): ...
     def client_setname(self, name): ...
+    def readwrite(self): ...
+    def readonly(self): ...
```
Comment (srittau) [inline third_party/2and3/redis/client.pyi]: As far as I can see from the implementation, those two commands return `bool`.

File: third_party/2and3/redis/client.pyi [hunk 2]
```diff
@@ -121,7 +123,7 @@ class Redis(object):
     __contains__: Any
     def expire(self, name: Union[Text, bytes], time: Union[int, timedelta]) -> bool: ...
     def expireat(self, name, when): ...
-    def get(self, name: Union[Text, bytes]) -> Optional[bytes]: ...
+    def get(self, name: Union[Text, bytes]): ...
```
Comment (srittau) [inline third_party/2and3/redis/client.pyi]: Why did you remove the return type here?

File: third_party/2and3/redis/client.pyi [hunk 3]
```diff
@@ -281,6 +283,7 @@ class Redis(object):
     def pubsub_numsub(self, *args: Text) -> List[Tuple[Text, int]]: ...
     def pubsub_numpat(self) -> int: ...
     def monitor(self) -> Monitor: ...
+    def cluster(self, cluster_arg, *args): ...
```
Comment (srittau) [inline third_party/2and3/redis/client.pyi]: `cluster_arg` must be a `str`. `args` can be annotated with `Any`, since we can't know the exact types (as they depend on the cluster command). Same is true for the return type, so this should also be `Any` to explicitly mark it.

File: third_party/2and3/redis/client.pyi (hunk 2 above)
Comment (brianmaissy) [reply third_party/2and3/redis/client.pyi]: sorry, i should have explained.

because it can return either a bytes or a string, depending on if the redis client class was initialized with `decode_repsonses=True`. from a typing perspective this is a bit of a problematic class, maybe it could be generic and overload the operator.

I decided the best solution for now is to not specify the return type at all, which is consistent with almost all of the other methods in the class.

File: third_party/2and3/redis/client.pyi (hunk 3 above)
Comment (brianmaissy) [reply third_party/2and3/redis/client.pyi]: about `cluster_arg` I agree, good point.

regarding the *args and return type, I understood (possibly incorrectly) the CONTRIBUTING.md as preferring unspecified parameter and return types rather than explicit `Any`s:

> Do not use Any to mark unannotated arguments or return values.
https://github.com/python/typeshed/blob/master/CONTRIBUTING.md#incomplete-stubs

Also I was copying the other annotations in the same file.

Would you prefer I used explicit `Any` in this case?

Comment (srittau) [reply third_party/2and3/redis/client.pyi]: Yes, no annotation is basically meant to show that no one has looked at the types yet, while `Any` means that someone has looked, but we can't use more precise types. The sentence from CONTRIBUTING does not explain that very well.

File: third_party/2and3/redis/client.pyi (hunk 2 above)
Comment (srittau) [reply third_party/2and3/redis/client.pyi]: Per my explanation above, we should use `Any` here then. We also often add a comment with the "real" return type in those cases as an additional hint to readers.
